ministryofjustice/analytics-platform-control-panel
# Feature/restore paramater store

## :memo: Summary
This PR contributes to https://github.com/ministryofjustice/analytical-platform/issues/3858

It reenables adding secrets to AWS Parameter Store for Airflow tasks only. Much of the code from the original implementation has been retained, with some changes to reflect it should only be used to add secrets related to Airflow tasks.

## :mag: What should the reviewer concentrate on?
- Code changes made to restrict use for Airflow only

## :technologist: How should the reviewer test these changes?
-An aws role is required to test with, you can create one in the ap-dev account, ensuring the name starts with "airflow". Or if it has not been deleted yet, you can use the one that I have created named `airflow-dev-example-role`
- Run the branch locally
- Go to the Parameters tab
- Create a new parameter, selecting the a valid role
- After submitting, go to AP-compute dev > Parameter Store and check that the secret has been added and looks correct. It should be the type "SecureString"
- Back in control panel, delete the parameter you created using the button on the parameter list page
- Check that it has been deleted as expected in the AP-dev AWS parameter store

## :books: Documentation status

- [x] Documentation [in the User Guidance](https://user-guidance.analytical-platform.service.justice.gov.uk/parameters.html) will be updated separately
